File: externals/Async-CTP/samples/CS/Windows/Phone/Concert/Finder/Concert/Finder/Converters/SearchableTypeConverter.cs
using System;
using System.Globalization;
using System.Windows.Data;
using ConcertFinder.Model;
using ConcertFinder.Resources;

namespace ConcertFinder.Converters
{
    /// <summary>
    /// Converter for ISearchable values.
    /// </summary>
    public class SearchableTypeConverter : IValueConverter
    {
        /// <summary>
        /// Convert an ISearchable to a string based on its type.
        /// </summary>
        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
        {
            var convertedValue = String.Empty;

            if (value is Artist)
            {
                convertedValue = Strings.Artist;
            }
            else if (value is Venue)
            {
                convertedValue = Strings.Venue;
            }

            return convertedValue;
        }

        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
        {
            throw new NotImplementedException();
        }
    }
}

File: externals/Async-CTP/samples/VB/Windows/Phone/Concert/Finder/Concert/Finder/Converters/SearchableTypeConverter.vb
Imports System
Imports System.Globalization
Imports System.Windows.Data
Imports Concert_Finder.ConcertFinder.Model

Namespace ConcertFinder.Converters

    ''' <summary>
    ''' Converter for ISearchable values.
    ''' </summary>
    Public Class SearchableTypeConverter
        Implements IValueConverter

        ''' <summary>
        ''' Convert an ISearchable to a string based on its type.
        ''' </summary>
        Public Function Convert(value As Object, targetType As [Type], parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.Convert
            Dim convertedValue = [String].Empty
            If TypeOf value Is Artist Then
                convertedValue = My.Resources.Strings.Artist
            ElseIf TypeOf value Is Venue Then
                convertedValue = My.Resources.Strings.Venue
            End If
            Return convertedValue
        End Function

        Public Function ConvertBack(value As Object, targetType As [Type], parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
            Throw New NotImplementedException()
        End Function

    End Class

End Namespace